src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -43,51 +42,211 @@ public ComplaintResponseService(ComplaintRepository complaintRepository, Complai
+     *
+     * @param complaintId id of the complaint for which to create an empty complaint response for
+     * @return persisted empty complaint response
+     */
+    public ComplaintResponse createEmptyComplaintResponse(Long complaintId) {
+        if (complaintId == null) {
+            throw new IllegalArgumentException("Complaint id should not be null");
+        }
+        Optional<Complaint> complaintFromDatabaseOptional = complaintRepository.findByIdWithEagerAssessor(complaintId);
+        if (complaintFromDatabaseOptional.isEmpty()) {
+            throw new IllegalArgumentException("Complaint was not found in database");
+        }
+        Complaint complaintFromDatabase = complaintFromDatabaseOptional.get();
+        if (complaintFromDatabase.isAccepted() != null) {
+            throw new IllegalArgumentException("Complaint is already handled");
+        }
+        if (complaintFromDatabase.getComplaintResponse() != null) {
+            throw new IllegalArgumentException("Complaint response already exists for given complaint");
+        }
+        Result originalResult = complaintFromDatabase.getResult();
         User assessor = originalResult.getAssessor();
-        User reviewer = this.userService.getUser();
+        User user = this.userService.getUser();
+        StudentParticipation studentParticipation = (StudentParticipation) originalResult.getParticipation();
+        if (!isUserAuthorizedToRespondToComplaint(studentParticipation, complaintFromDatabase, assessor, user)) {
+            throw new AccessForbiddenException("Insufficient permission for creating the empty complaint response");
+        }
+        ComplaintResponse emptyComplaintResponse = new ComplaintResponse();
+        emptyComplaintResponse.setReviewer(user); // owner of the lock
+        emptyComplaintResponse.setComplaint(complaintFromDatabase);
+        ComplaintResponse persistedComplaintResponse = complaintResponseRepository.save(emptyComplaintResponse);
+        log.debug("Created empty complaint and thus lock for complaint with id : {}", complaintId);
+        return persistedComplaintResponse;
+    }
 
+    /**
+     * Resolves a complaint by filling in the empty complaint response attached to it
+     *
+     * The empty complaint response acts as a lock. Only the creator of the empty complaint response and instructors can resolve empty complaint response as long as the lock
+     * is running. For lock duration calculation see: {@link ComplaintResponse#isCurrentlyLocked()}. This methods fill in the initial complaint response and either accepts
+     * or denies the associated complaint, thus resolving the complaint
+     *
+     * @param updatedComplaintResponse complaint response containing the information necessary for resolving the complaint
+     * @return complaintResponse of resolved complaint
+     */
+    @Transactional // ok because of modifying query
```

Comment:
please remove `@Transactional`

Reply:
Done in 8e3e3774528e1dc63f6450e98a65532ae01f7f06